pyramid_tm/__init__.py

```diff
@@ -126,4 +136,18 @@ def includeme(config):
     - If none of the above conditions are True, the transaction will be
       committed (via ``transaction.commit()``).
     """
+    config.add_request_method(
+        'pyramid_tm.create_tm', name='transaction', reify=True,
+    )
     config.add_tween('pyramid_tm.tm_tween_factory', under=EXCVIEW)
+
+    def ensure():
+        manager_hook = config.registry.settings.get("tm.manager_hook")
+        if manager_hook is not None:
+            try:
+                manager_hook = resolver.maybe_resolve(manager_hook)
+                config.registry.settings["tm.manager_hook"] = manager_hook
+            except (ImportError, ValueError):
+                raise ConfigurationError("Unable to resolve tm.manager_hook")
+
+    config.action(None, ensure, order=10)
```

Comment:
Actually my settings not taking effect would surprise me more, especially since I didn't call `commit()` either explicitly or implicitly. I would expect that it would use my settings from that `commit()` cycle, and not from before then, ignoring any changes I've made.

Although, that is based upon all of the crap we got when we modified `pyramid_mako` to do exactly what you are advocating here...